### DISCUSSION TITLE
new installation deleted half my files till microsoft locked client

### DISCUSSION THREAD

**Discussion by switch7derek** [abraunegg/onedrive#1415] [Q&A] [answered]
hi, i've been using this product on a number of machines for a number of years.  i just did a new build, linux xubuntu, and installed onedrive and then did onedrive --synchronize.  after watching things get started normally i walked away.  when i returned the last message on the screen was: ERROR: An attempt to remove a large volume of data from OneDrive has been detected. Exiting client to preserve data on OneDrive
ERROR: To delete delete a large volume of data use --force or increase the config value 'classify_as_big_delete' to a larger value
thinking to myself, "naaaa, that couldn't have really just happened", i log into my account via the web interface, and sure enough, i have lost at least half of my repository.  i have no idea why this happened.  the only thing i can think that was different with this build/install is that i put the root onedrive folder on a m2 drive mounted at /opt, where as on all my other installs i've gone with the default ~/OneDrive.
i am not raising this hoping to get my files back ( i do have them backed up offline ), but, if this wasn't a one time anomaly, well, even if it was, it's a pretty big deal - and i figured i should raise it up to the powers that be.  i'd also like it to not happen the next time i install onedrive.  :-)

**Comment by switch7derek** [comment]
sorry, i suppose relevant would be that i installed from the instructions:

Install via the Ubuntu PPA Archive:
```
sudo add-apt-repository ppa:yann1ck/onedrive
sudo apt-get update
sudo apt install onedrive
```

**Comment by abraunegg** [comment, marked as answer]
@switch7derek 
> i have no idea why this happened

This error occurred because you used the Ubuntu PPA package and were hit by https://github.com/abraunegg/onedrive/issues/1274 - because the PPA automatically starts the client - which it is *not* supposed to do - so the moment you authenticated the client, it started syncing using the default configuration. When you then updated your configuration, the client thought your files were *gone* thus delete.

> i am not raising this hoping to get my files back ( i do have them backed up offline )

If you look on OneDrive online, your files *should* be in the recycle bin there to restore. Before doing this, ensure you remove the Ubuntu PPA autostart systemd file - see https://github.com/abraunegg/onedrive/issues/1274 on how to do this

> i'd also like it to not happen the next time

Make sure you *always* test your configuration with `--dry-run` before activating any systemd service .. yes in this case you had no idea probably that it was already running - but this is why https://github.com/abraunegg/onedrive/issues/1274 exists to let folk know this is an issue - but, as I do not do the packaging of the applications, cannot I fix that.

Additionally, as you are mounting to a different folder, incorrect permissions can also cause this issue. What you can also do is use a `.nosync` file in the mount point, update your configuration so that if the mount is never there, the sync aborts much earlier:
```
--check-for-nomount
      Check for the presence of .nosync in the syncdir root. If found, do not perform sync.
```

Things for you do do:
1. Upgrade your client - v2.4.10 from the Ubuntu PPA is not supported. There are 16+ issues that have been fixed since that release.
2. Remove the Ubuntu PPA package autostart configuration as per https://github.com/abraunegg/onedrive/issues/1274
3. Implement `.nosync` for the mount point and implement mount point check
4. Test with `--dry-run` before activating any systemd service

**Reply by switch7derek** [reply]
ah, i see.  thanks so much for the explanations.  will take note and follow your recommendations.
cheers!
